app/controllers/assessments_controller.rb

```diff
@@ -0,0 +1,69 @@
+class AssessmentsController < ApplicationController
+  before_action :set_contact, only: %i[new create]
+
+  def new
+    @users = User.all
+    @roles = Role.all
+    @type = %w[log schedule].include?(type_param) ? type_param : 'log'
+
+    @need = Need.new
+    @note = Note.new
+
+    @need.status = 'complete' if @type == 'log'
+    @need.user = current_user if @type == 'log'
+  end
+
+  def create
+    @type = %w[log schedule].include?(type_param) ? type_param : 'log'
+    if @type == 'log'
+      log_assessment
+    else
+      schedule_assessment
+    end
+  end
+
+  private
+
+  def set_contact
+    @contact = Contact.find(params[:contact_id])
+  end
+
+  def log_assessment
+    @need = Need.new(assessment_params.merge(contact_id: @contact.id, name: 'Logged Assessment'))
+    @note = Note.new(notes_params.merge(need: @need))
```

Comment:
Interesting: the `need_id` of the note must come from a hidden html element or something, as opposed to associating them here? 🤔